src/Workspaces/MSBuild/BuildHost/BuildHost.cs

```diff
@@ -59,14 +60,15 @@ private bool TryEnsureMSBuildLoaded(string projectOrSolutionFilePath)
 
                 // Locate the right SDK for this particular project; MSBuildLocator ensures in this case the first one is the preferred one.
                 // TODO: we should pick the appropriate instance back in the main process and just use the one chosen here.
-                var options = new VisualStudioInstanceQueryOptions { DiscoveryTypes = DiscoveryType.DotNetSdk, WorkingDirectory = Path.GetDirectoryName(projectOrSolutionFilePath) };
+                var options = new VisualStudioInstanceQueryOptions { DiscoveryTypes = DiscoveryType.DotNetSdk, WorkingDirectory = Path.GetDirectoryName(projectOrSolutionFilePath), AllowAllDotnetLocations = true, AllowAllRuntimeVersions = true };
```

Comment:
Discover all SDKs from all dotnet locations regardless of the runtime we are running on. This will allow us to run from a local 9.0.x runtime installed by the .NET Install Tool and still discover user installed .NET 10.0 preview SDKs.

Reply:
This might be worth adding this comment into the code in some form.